pkg/webservice/handlers.go

```diff
@@ -418,6 +420,80 @@ func getNodesUtilJSON(partition *scheduler.PartitionContext, name string) *dao.N
 	}
 }
 
+func getNodeUtilisations(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	var result []*dao.PartitionNodesUtilDAOInfo
+	for _, part := range schedulerContext.GetPartitionMapClone() {
+		partitionNodesUtilJSON := getPartitionNodesUtilJSON(part)
+		result = append(result, partitionNodesUtilJSON)
+	}
+
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		buildJSONErrorResponse(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// getPartitionNodesUtilJSON retrieves the utilization of all resource types for nodes within a specific partition.
+func getPartitionNodesUtilJSON(partition *scheduler.PartitionContext) *dao.PartitionNodesUtilDAOInfo {
+	type UtilizationBucket struct {
+		NodeCount []int      // 10 buckets, each bucket contains number of nodes
+		NodeList  [][]string // 10 buckets, each bucket contains node name list
+	}
+	resourceBuckets := make(map[string]*UtilizationBucket) // key is resource type, value is UtilizationBucket
+
+	// put nodes to buckets
+	for _, node := range partition.GetNodes() {
+		capacity := node.GetCapacity()
+		resourceAllocated := node.GetAllocatedResource()
+		absUsedCapacity := resources.CalculateAbsUsedCapacity(capacity, resourceAllocated)
+
+		// append to bucket based on resource type, only count if node advertises the resource
+		for resourceType := range capacity.Resources {
+			idx := 0
+			if absValue, ok := absUsedCapacity.Resources[resourceType]; ok {
```

Comment:
what if the keys of `absUsedCapacity` are not equal to `capacity`? the `idx` is always 0

Reply:
That's as expected.
If a key existing in capacity but not in absUsedCapacity(Not used before),  the node should be count in 0%-10% bucket. (idx=0)

ex:
If node have GCP/TPU, but never used those resource before.  
The absUsedCapacity won't have GCP/TPU abs value.

So we will count the node to
- resourceBuckets["GPU"].NodeCount[0]++
- resourceBuckets["CPU"].NodeCount[0]++